### DISCUSSION TITLE
New PDS CLI Guidelines?

### DISCUSSION THREAD

**Discussion by jordanpadams** [NASA-PDS/PDS-Software-Issues-Repo#29] [General]
New proposed CLI Guidelines for PDS. Thoughts?

https://clig.dev/

**Comment by tloubrieu-jpl** [comment]
I would like that the conclusion of the discussion gives us recommendation on which library to use to parse command line arguments.

There is a list of them proposed there https://clig.dev/#the-basics

We need to assess that they are consistent from a user point of view. The main ones for us would be picocli for java (we recommended one once to @jshughes, hopefully that was the one) and argparse for python. I've also seen Click for python used for another project I am working on , I guess it is similar to argparse from a user point of view but it sounds like it is more powerful, see https://click.palletsprojects.com/en/8.0.x/why/#why-not-argparse . I believe that can propose existing options when the user have typos.

**Comment by jshughes** [comment]
@tloubrieu-jpl As mentioned, there were several suggested for IMTool/LDDTool. The one implemented was argparse4j-0.9.0.jar. If necessary it could be changed.
